kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/encrypt/InBandEncryptionManager.java

```diff
@@ -114,9 +118,9 @@ private CompletionStage<MemoryRecords> attemptEncrypt(@NonNull String topicName,
                             + partition + " after " + attempt + " attempts"));
         }
         return currentDek(encryptionScheme).thenCompose(dek -> {
-            // if it's not alive we know a previous encrypt call has removed this stage from the cache and fall through to retry encrypt
-            if (!dek.isDestroyed()) {
-                try (Dek<E>.Encryptor encryptor = dek.encryptor(allRecordsCount)) {
+            Dek<E> eDek = dek.getDek();
+            if (!eDek.isDestroyed() && !eDek.destroyedForEncrypt()) {
```

Comment:
```suggestion
            if (!eDek.destroyedForEncrypt()) {
```
`destroyedForEncrypt` should be a strict subset of values for `isDestroyed` right? So I think we should simplify the condition.

However that also makes me think we have the API backwards. Instead of `isDestroyed` it should be `canEncrypt` or `canDecrypt` or maybe `canUseFor(ENCRYPTION)`

Reply:
or maybe we should keep the attempt-to-use-and-throw pattern to have less places where we consult the state. It just felt a little wrong to use exceptions if we know we can't use it due to destruction